sentinel-core/src/main/java/com/alibaba/csp/sentinel/slots/system/SystemStatusListener.java

```diff
@@ -54,7 +60,21 @@ public void run() {
              * observed. All values between 0.0 and 1.0 are possible depending of the activities going on in the
              * system. If the system recent cpu usage is not available, the method returns a negative value.
              */
-            currentCpuUsage = osBean.getSystemCpuLoad();
+            double systemCpuUsage = osBean.getSystemCpuLoad();
+
+            // calculate process cpu usage to support application running in container environment
+            RuntimeMXBean runtimeBean = ManagementFactory.getPlatformMXBean(RuntimeMXBean.class);
+            long newProcessCpuTime = osBean.getProcessCpuTime();
+            long newProcessUpTime = runtimeBean.getUptime();
```

Comment:
What's the difference between `uptime` and `systemtime` ? Avoid unexpected changing of system time?

Reply:
uptime won't be affected by unexpected changing of system time, the mechanism is similar to System.nanoTime()

Reply:
> `uptime` won't be affected by unexpected changing of system time, the mechanism is similar to `System.nanoTime()`

Yeah it's okay for me if no portable issue on RuntimeMXBean